src/power_grid_model/_core/power_grid_core.py

```diff
@@ -125,11 +126,26 @@ def _load_core() -> CDLL:
     Returns: DLL/SO object
 
     """
-    if platform.system() == "Windows":
-        dll_file = "_power_grid_core.dll"
+    if "CONDA_PREFIX" in os.environ:
+        # if conda environment, use the dll file from the conda environment
+        if platform.system() == "Windows":
+            dll_file = "power_grid_model_c.dll"
+        elif platform.system() == "Darwin":
+            dll_file = "libpower_grid_model_c.dylib"
+        elif platform.system() == "Linux":
+            dll_file = "libpower_grid_model_c.so"
+        else:
+            raise NotImplementedError(f"Unsupported platform: {platform.system()}")
+        # the dll will be found through conda environment
+        dll_path = dll_file
     else:
```

Comment:
this is potentially breaking for users who use conda as their base environment (e.g. because they installed python via conda) and run both conda and venv next to eachother. i think we should first look for a local file of PGM and then change to the global version.

Especially people installing PGM themselves may accidentally load the PGM version from outside the current working directory